gateway/gateway-requests/src/registration/handshake/gateway.rs

```diff
@@ -48,12 +50,29 @@ impl<'a> GatewayHandshake<'a> {
                     }
                 }
 
-                // init: <- pub_key || g^x
-                let (remote_identity, remote_ephemeral_key) = check_processing_error(
+                // init: <- pub_key || g^x || credential
```

Comment:
I wonder whether there might be some attack surface here if we're sending the credential in plain that is not bound to anyone. Perhaps it should be sent towards the end of the handshake so that it'd be encrypted with the derived shared key? I honestly don't know. Perhaps @aniampio might have an opinion

Reply:
As per the offline discussion, I'll do the credential send after the handshake, so that it's protected with the shared key.